staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -234,6 +243,12 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 		c.GenericConfig.SharedInformerFactory.Start(context.StopCh)
 		return nil
 	})
+	s.GenericAPIServer.AddPostStartHookOrDie("start-kube-aggregator-dynamic-cert-reload", func(context genericapiserver.PostStartHookContext) error {
```

Comment:
how about registering the post start hook only if `certKeyContentProvider != nil`